src/main/java/io/seqera/tower/cli/shared/WorkflowDumpExportFormat.java

```diff
@@ -0,0 +1,190 @@
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import io.seqera.tower.JSON;
+import io.seqera.tower.cli.commands.runs.DumpCmd;
+import io.seqera.tower.model.Workflow;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * This is the class used by {@link DumpCmd} to structure Workflow data as JSON.
+ * The class {@link Workflow} is missing some extra fields so this envelope
+ * class is needed to export/import them along the rest of the data.
+ */
+public final class WorkflowDumpExportFormat {
+
+    private Workflow workflow;
+
+    private Long pipelineId;
+
+    private Long workspaceId;
+
+    private Long userId;
+
+    private String userEmail;
+
+    public WorkflowDumpExportFormat(
+            final Workflow workflow,
+            final Long pipelineId,
+            final Long workspaceId,
+            final Long userId,
+            final String userEmail
+    ) {
+        this.workflow = workflow;
+        this.pipelineId = pipelineId;
+        this.workspaceId = workspaceId;
+        this.userId = userId;
+        this.userEmail = userEmail;
+    }
+
+    public Workflow getWorkflow() {
+        return workflow;
+    }
+
+    public Long getPipelineId() {
+        return pipelineId;
+    }
+
+    public Long getWorkspaceId() {
+        return workspaceId;
+    }
+
+    public Long getUserId() {
+        return userId;
+    }
+
+    public String getUserEmail() {
+        return userEmail;
+    }
+
+    public static WorkflowDumpExportFormat deserialize(final String json) throws JsonProcessingException {
+        return buildMapper()
+                .registerModule(
+                        new SimpleModule()
+                                .addDeserializer(WorkflowDumpExportFormat.class, new WorkflowDumpExportFormatDeserializer())
+                )
+                .readValue(json, WorkflowDumpExportFormat.class);
+    }
+
+    /**
+     * This mixin class allows to add a virtual fields to {@link Workflow}.
+     * See attribute inclusion in {@link WorkflowDumpExportFormat#serialize(WorkflowDumpExportFormat)}
+     */
+    @JsonAppend(attrs = {
+            @JsonAppend.Attr(value = "pipelineId"),
+            @JsonAppend.Attr(value = "workspaceId"),
+            @JsonAppend.Attr(value = "userId"),
+            @JsonAppend.Attr(value = "userEmail"),
+    })
+    public static abstract class WorkflowDumpMixin {}
+
+    public static String serialize(final WorkflowDumpExportFormat wfExport) throws JsonProcessingException {
+        return buildMapper()
+                .writerWithDefaultPrettyPrinter()
+                .withAttribute("pipelineId", wfExport.getPipelineId())
+                .withAttribute("workspaceId", wfExport.getWorkspaceId())
+                .withAttribute("userId", wfExport.getUserId())
+                .withAttribute("userEmail", wfExport.getUserEmail())
+                .writeValueAsString(wfExport.getWorkflow());
```

Comment:
Thought: related to @jordeu 's point above: What happens if in the future `Workflow` class will have attributes with the same name as the envelope class `WorkflowDumpExportFormat`, like `pipelineId` present in both?

Probably both `pipelineId` attributes will mean the same thing, they won't be different, so that might not be an issue. 

But I'm curios what will be the behavior of the serializer. Will it throw an error - duplicate name in JSON? Or will it overwrite, the later write (for `Workflow`) will overwrite the former write (for `wfExport`)? It could be helpful to have a unit test verifying this behavior.

Reply:
Yeah, this is risky but unfortunately there is no other way of adding "virtual" fields to JSON.
We are going to separate the added metadata into another file so hopefully that will make this future-proof.